miio/viomivacuum.py

```diff
@@ -423,22 +809,165 @@ def set_dnd(
+
+        [{'name': 'MapName1', 'id': 1598622255, 'cur': False},
+         {'name': 'MapName2', 'id': 1599508355, 'cur': True},
+          ...]
+        """
+        if not self._cache["maps"]:
+            self._cache["maps"] = self.send("get_map")
+        return self._cache["maps"]
+
+    @command(click.argument("map_id", type=int))
+    def set_map(self, map_id: int):
+        """Change current map."""
+        maps = self.get_maps()
+        if map_id not in [m["id"] for m in maps]:
+            raise ViomiVacuumException("Map id {} doesn't exists".format(map_id))
+        return self.send("set_map", [map_id])
+
+    @command(click.argument("map_id", type=int))
+    def delete_map(self, map_id: int):
+        """Delete map."""
+        maps = self.get_maps()
+        if map_id not in [m["id"] for m in maps]:
+            raise ViomiVacuumException("Map id {} doesn't exists".format(map_id))
+        return self.send("del_map", [map_id])
+
+    @command(
+        click.argument("map_id", type=int),
+        click.argument("map_name", type=str),
+    )
+    def rename_map(self, map_id: int, map_name: str):
+        """Rename map."""
+        maps = self.get_maps()
+        if map_id not in [m["id"] for m in maps]:
+            raise ViomiVacuumException("Map id {} doesn't exists".format(map_id))
+        return self.send("rename_map", {"mapID": map_id, "name": map_name})
+
+    @command(
+        click.option("--map-id", type=int, default=None),
+        click.option("--map-name", type=str, default=None),
+        click.option("--refresh", type=bool, default=False),
+    )
+    def get_rooms(
+        self, map_id: int = None, map_name: str = None, refresh: bool = False
+    ):
+        """Return room ids and names."""
+        if self._cache["rooms"] and not refresh:
+            return self._cache["rooms"]
+        if map_name:
+            map_id = None
+            maps = self.get_maps()
+            for map_ in maps:
+                if map_["name"] == map_name:
+                    map_id = map_["id"]
+            if map_id is None:
+                raise ViomiVacuumException(
+                    "Error: Bad map name, should be in {}".format(
+                        ", ".join([m["name"] for m in maps])
+                    )
+                )
+        elif map_id:
+            maps = self.get_maps()
+            if map_id not in [m["id"] for m in maps]:
+                raise ViomiVacuumException(
+                    "Error: Bad map id, should be in {}".format(
+                        ", ".join([str(m["id"]) for m in maps])
+                    )
+                )
+        # Get scheduled cleanup
+        schedules = self.send("get_ordertime", [])
+        scheduled_found, rooms = _get_rooms_from_schedules(schedules)
+        if not scheduled_found:
+            msg = (
+                "Fake schedule not found. "
+                "Please create a scheduled cleanup with the "
+                "following properties:\n"
```

Comment:
Is it always necessary to have this "fake schedule"? If not, maybe just log a warning saying that there are no scheduled cleans? Otherwise `get_rooms` would be unusable due to the exception.

Reply:
Yes it's mandatory to have schedules in order to see the rooms.
So what do you want me to do ?

Reply:
Ah, okay. I assume there is no way to such a fake schedule programatically (at least for)? Assuming any schedule (and not just a fake one) is fine, the error message should rather be something like "No schedules found. You can create a fake schedule with the following properties ..", but I'm okay with this for now.